uai2017/python/metadata-papers.py

```diff
@@ -0,0 +1,141 @@
+import argparse
+import openreview
+from collections import defaultdict
+import xml.etree.cElementTree as ET
+import xml.dom.minidom as minidom
+from uaidata import *
+
+# Parse the arguments for user authentication
+# .............................................................................
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--baseurl', help="base URL")
+parser.add_argument('--username')
+parser.add_argument('--password')
+parser.add_argument('--bidscores', help="The xml file containing the reviewer bids")
+parser.add_argument('--out', help="output file")
+
+args = parser.parse_args()
+
+if args.username is not None and args.password is not None:
+    client = openreview.Client(baseurl=args.baseurl, username=args.username, password=args.password)
+else:
+    client = openreview.Client(baseurl=args.baseurl)
+
+
+# Function definitions
+# .............................................................................
+
+def prettify(elem):
+    """Return a pretty-printed XML string for the Element.
+    """
+    rough_string = ET.tostring(elem, 'utf-8')
+    reparsed = minidom.parseString(rough_string)
+    return reparsed.toprettyxml(indent="\t")
+
+
+# Organize data
+# .............................................................................
+
+bid_score_map = {
+ 'I want to review': 1.0,
+ 'I can review': 0.75,
+ 'I can probably review but am not an expert': 0.5,
+ 'I cannot review': 0.25,
+ 'No bid': 0.0
+}
+
+reviewers = client.get_group(PC)
+bids = client.get_tags(invitation='auai.org/UAI/2017/-/Add/Bid')
+metadata_notes = client.get_notes(invitation = 'auai.org/UAI/2017/-/Paper/Metadata')
+metadata_by_id = {n.forum:n for n in metadata_notes}
+
+bids_by_number = defaultdict(list)
+bids_by_id = defaultdict(list)
+
+for b in bids:
+    n = client.get_note(b.forum)
+    bids_by_number[n.number].append(b)
+    bids_by_id[n.forum].append(b)
+
+
+recs = client.get_tags(invitation='auai.org/UAI/2017/-/Recommend/Reviewer')
```

Comment:
I changed the invitation, now we have one invitation per paper:

auai.org/UAI/2017/-/Paper.*/Recommend/Reviewer